test/IonBinaryWriter.ts

```diff
@@ -0,0 +1,1139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+import {assert} from 'chai';
+import * as ion from '../src/IonTests';
+
+const ivm = [0xe0, 0x01, 0x00, 0xea];
+
+let writerTest = function(name, instructions, expected) {
+    it(name, () => {
+        let symbolTable = new ion.LocalSymbolTable(ion.getSystemSymbolTableImport());
+        let writeable = new ion.Writeable();
+        let writer = new ion.BinaryWriter(symbolTable, writeable);
+        instructions(writer);
+        writer.close();
+        let actual = writeable.getBytes();
+        assert.deepEqual(actual, new Uint8Array(ivm.concat(expected)));
+    });
+}
+
+let badWriterTest = function(name, instructions) {
+    let test = () => {
+        let symbolTable = new ion.LocalSymbolTable(ion.getSystemSymbolTableImport());
+        let writeable = new ion.Writeable();
+        let writer = new ion.BinaryWriter(symbolTable, writeable);
+        instructions(writer);
+        writer.close();
+    };
+    it(name, () => assert.throws(test, Error));
+}
+
+let blobWriterTests = [
+    {
+        name: "Writes blob",
+        instructions: (writer) => writer.writeBlob(new Uint8Array([1, 2, 3])),
+        expected: [0xa3, 1, 2, 3]
+    },
+    {
+        name: "Writes null blob 1",
+        instructions: (writer) => writer.writeBlob(null),
+        expected: [0xaf]
+    },
+    {
+        name: "Writes null blob 2",
+        instructions: (writer) => writer.writeNull(ion.IonTypes.BLOB),
+        expected: [0xaf]
+    },
+    {
+        name: "Writes blob with annotation",
+        instructions: (writer) => {
+            writer.setAnnotations(['a']);
+            writer.writeBlob(new Uint8Array([1]));
+        },
+        expected: [
+            // '$ion_symbol_table'::
+            0xe7,
+            0x81,
+            0x83,
+            // {
+            0xd4,
+            // symbols:
+            0x87,
+            // [
+            0xb2,
+
+            // "a"
+            0x81,
+            'a'.charCodeAt(0),
+
+            // ] }
+
+            // Annotations
+            0xe4,
+            0x81,
+            0x8a,
+
+            // Blob
+            0xa1,
+            1,
+        ]
+    },
+];
+
+let booleanWriterTests = [
+    {
+        name: "Writes boolean true",
+        instructions: (writer) => writer.writeBoolean(true),
+        expected: [0x11]
+    },
+    {
+        name: "Writes boolean false",
+        instructions: (writer) => writer.writeBoolean(false),
+        expected: [0x10]
+    },
+    {
+        name: "Writes null boolean by detecting null",
+        instructions: (writer) => writer.writeBoolean(null),
+        expected: [0x1f]
+    },
+    {
+        name: "Writes null boolean by detecting undefined",
+        instructions: (writer) => writer.writeBoolean(undefined),
```

Comment:
should we open an issue here tracking removal undefined from the boolean api?

Reply:
#456 covers this.